homeassistant/components/arris_tg2492lg/device_tracker.py

```diff
@@ -40,23 +51,23 @@ def __init__(self, connect_box: ConnectBox) -> None:
         self.connect_box = connect_box
         self.last_results: list[Device] = []
 
-    def scan_devices(self) -> list[str]:
+    async def async_scan_devices(self) -> list[str]:
         """Scan for new devices and return a list with found device IDs."""
-        self._update_info()
+        await self._async_update_info()
 
         return [device.mac for device in self.last_results if device.mac]
 
-    def get_device_name(self, device: str) -> str | None:
+    async def async_get_device_name(self, device: str) -> str | None:
```

Comment:
Was it necessary to make this function async? I couldn't see anything that needed this.

Reply:
A coroutine (as in this PR) is preferred over a sync method because the latter will be handed to an executor.